tests/unit/task/test_run.py

```diff
@@ -50,3 +59,72 @@ def test_run_task_preserve_edges():
         task.get_graph_queue()
         # when we get the graph queue, preserve_edges is True
         mock_node_selector.get_graph_queue.assert_called_with(mock_spec, True)
+
+
+class TestModelRunner:
+    @pytest.fixture
+    def log_model_result_catcher(self) -> EventCatcher:
+        return EventCatcher(event_to_catch=LogModelResult)
+
+    @pytest.fixture
+    def model_runner(
+        self,
+        postgres_adapter: PostgresAdapter,
+        table_model: ModelNode,
+        runtime_config: RuntimeConfig,
+    ) -> ModelRunner:
+        return ModelRunner(
+            config=runtime_config,
+            adapter=postgres_adapter,
+            node=table_model,
+            node_index=1,
+            num_nodes=1,
+        )
+
+    @pytest.fixture
+    def run_result(self, table_model: ModelNode) -> RunResult:
+        return RunResult(
+            status=RunStatus.Success,
+            timing=[],
+            thread_id="an_id",
+            execution_time=0,
+            adapter_response={},
+            message="It did it",
+            failures=None,
+            node=table_model,
+        )
+
+    def test_print_result_line(
+        self,
+        log_model_result_catcher: EventCatcher,
+        model_runner: ModelRunner,
+        run_result: RunResult,
+    ) -> None:
+        # Setup way to catch events
+        event_manager = get_event_manager()
+        event_manager.callbacks.append(log_model_result_catcher.catch)
+
+        # Check `print_result_line` with "successful" RunResult
+        model_runner.print_result_line(run_result)
+        assert len(log_model_result_catcher.caught_events) == 1
+        assert log_model_result_catcher.caught_events[0].info.level == EventLevel.INFO
+        assert log_model_result_catcher.caught_events[0].data.status == run_result.message
+
+        # reset event catcher
+        log_model_result_catcher.flush()
+
+        # Check `print_result_line` with "error" RunResult
+        run_result.status = RunStatus.Error
+        model_runner.print_result_line(run_result)
+        assert len(log_model_result_catcher.caught_events) == 1
+        assert log_model_result_catcher.caught_events[0].info.level == EventLevel.ERROR
+        assert log_model_result_catcher.caught_events[0].data.status == EventLevel.ERROR
+
+    @pytest.mark.skip(
+        reason="Default and adapter macros aren't being appropriately populated, leading to a runtime error"
+    )
+    def test_execute(
+        self, table_model: ModelNode, manifest: Manifest, model_runner: ModelRunner
+    ) -> None:
+        model_runner.execute(model=table_model, manifest=manifest)
```

Comment:
Any thoughts on how we can actually test this? I think we probably want to think a bit more about what the `runner` layer logic is? Is it run a specific macro with correct parameters? or something else

Reply:
A non-exhaustive list of things we could unit test about the `execute` function
1. The result on success
  a. `execute` should return a `RunResult`
    1. assert that the status is `success`
    2. assert that the `adapter_response` is as expected
2. If the `materialization_macro` is not found, the appropriate error is raised
  a. this is what we are unintentionally doing currently (and we want to be doing it intentionally)
3. If `config` is not in `context` the appropriate error is raised
4. If the materialization macro doesn't support the language of the model, the appropriate error is raised
5. Pre model hook is fired
6. Post model hook is fired

What is blocking most of this currently is (2.a)

Reply:
Should we open a follow up ticket about 2.a?

Reply:
Yes indeededy 👍